### PR TITLE
Fixes #11437 - avoiding to modify frozen arrays

### PR DESCRIPTION
**Pull request by xprazak2**
We check if array is frozen, then modify a duplicate and freeze again

### CODE REVIEW

File: app/helpers/layout_helper.rb [hunk 1]
```diff
@@ -122,7 +122,13 @@ def select_f(f, attr, array, id, method, select_options = {}, html_options = {})
     if include_blank
       blank_value = include_blank.is_a?(TrueClass) ? nil : include_blank
       blank_option = OpenStruct.new({id => '', method => blank_value })
+      frozen = false
```
Comment (ShimShtein) [inline app/helpers/layout_helper.rb]: Good catch, but IMHO it's better to clone the array in any case, and not just if it's frozen. This way we will avoid modification of the source, which may lead to pretty unexpected side effects.
The problem is half-solved on the first line of the method, but for the case of arrays `to_a` returns `self`. We just need to treat it there.

**Comment by xprazak2** [comment]
Now we duplicate the array before calling `to_a`, so we do not modify the original object at all.

File: app/helpers/layout_helper.rb [hunk 2]
```diff
@@ -113,7 +113,7 @@ def radio_button_f(f, attr, options = {})
   end
 
   def select_f(f, attr, array, id, method, select_options = {}, html_options = {})
-    array = array.to_a
+    array = array.dup.to_a
```
Comment (domcleal) [inline app/helpers/layout_helper.rb]: Nitpick, I think duplicating the array (after to_a) may be clearer, as at least we always know that it's duping an array and not whatever array-like object `array` happens to be.

**Comment by xprazak2** [comment]
We are duplicating the array object.

**Comment by xprazak2** [comment]
[test]

**Comment by shlomizadok** [comment]
Tested. Seems good to merge

**Comment by domcleal** [comment]
Merged as b8fa2e2, thanks @xprazak2.

**Pull request closed without merging**